src/deprecation.cpp

```diff
@@ -27,6 +28,12 @@ bool fEnableWalletTxVJoinSplit = true;
 
 static const std::string CLIENT_VERSION_STR = FormatVersion(CLIENT_VERSION);
 
+int64_t EstimatedNodeDeprecationTime(const CClock& clock, int nHeight) {
+    auto blocksToDeprecation = DEPRECATION_HEIGHT - nHeight;
+
+    return clock.GetTime() + (blocksToDeprecation * Consensus::POST_BLOSSOM_POW_TARGET_SPACING);
```

Comment:
If `nHeight` is after the EoS height, this will get increasingly inaccurate as time goes on. Perhaps it should be the time at which the EoS height actually occurred on the current best chain, in that case?

Reply:
We don't track that anywhere at present, do we?

Reply:
Once the EoS height is reached, the node shuts down and refuses to restart. So it's unlikely that anyone can query the RPC at that point. (I don't recall if it might be possible to get an RPC query in before the restart detects it shouldn't be running, but even then it wouldn't be with any larger `nHeight` because no new blocks would get downloaded.)

Reply:
As pointed out by @str4d, we can't query the RPC after the EoS height. I was confused by this in the RPC doc: https://github.com/zcash/zcash/pull/6425/files#diff-2ae094f9329ce9498e1b4a9cb7767945b54a33b4ee9fed4f0ba401a98e1683c1R445-R446